provider/testdata/src/lib.rs

```diff
@@ -242,3 +242,13 @@ mod baked {
     include!(concat!(env!("CARGO_MANIFEST_DIR"), "/data/baked/mod.rs"));
     include!(concat!(env!("CARGO_MANIFEST_DIR"), "/data/baked/any.rs"));
 }
+
+const _: () = {
+    use crate::baked::BakedDataProvider;
+    use icu_provider_adapters::fallback::provider::*;
+    icu_provider::impl_dynamic_data_provider!(
```

Comment:
Can you please summarise what's going on here?

Reply:
See above for why I need a DynamicDataProvider impl.

I added it in a private module so that the imports I need are more localized.